extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/scan/ScanQueryFrameProcessor.java

```diff
@@ -172,13 +174,26 @@ public static Sequence<Object[]> mappingFunction(Sequence<ScanResultValue> input
     }).map(List::toArray);
   }
 
+  private static ScanQuery prepareScanQuery(@NotNull ScanQuery scanQuery)
```

Comment:
Can you please add a Javadoc for how we are preparing the scan query.

Reply:
Added